app/controllers/comments_controller.rb

```diff
@@ -1,24 +1,32 @@
 class CommentsController < ApplicationController
   def new
-    @new_comment = Comment.new
+    @comment = Comment.new
   end
 
   def create
-    # @user = User.find(params[:user_id])
     @post = Post.find(params[:post_id])
-    @new_comment = @post.comments.new(comment_params)
-    @new_comment.author_id = @post.id
-    @new_comment.post_id = params[:post_id]
+    @new_comment = @post.comments.new(
+      text: comment_params,
+      author_id: @post.id,
+      post_id: @post.id
+    )
+    @new_comment.post_id = @post.id
     @new_comment.update_comments_counter
-
-    if @new_comment.save
-      redirect_to user_post_path(@post.id, Post.find(params[:post_id]))
-    else
-      render :new
+    respond_to do |format|
+      format.html do
+        if @new_comment.save
+          redirect_to "/users/#{@post.author_id}/posts/#{@post.id}", notice: 'Success!'
+          # redirect_to user_post_path(@post.id, Post.find(params[:post_id]))
+        else
+          render :new, alert: 'Failed to Create!'
+        end
+      end
```

Comment:
- [ ] Still, the comment creation method is not working. As you can see `@new_comment.save` is not false in the if statement so it is directly moved to the else statement. Please fix it.

![image](https://user-images.githubusercontent.com/38635158/160708716-a8e26523-602e-40ea-85e0-2423049dd094.png)